src/main/bash/sdkman-upgrade.sh

```diff
@@ -39,11 +39,14 @@ function __sdk_upgrade() {
 		upgradable="$(__sdkman_determine_upgradable_version "$candidate")"
 		case $? in
 		1)
-			$all || __sdkman_echo_red "Not using any version of ${candidate}"
+			$all || {
+				__sdkman_echo_stop "Not using any version of ${candidate}"
+				return 1
+			}
```

Comment:
I added a "Stop!" here. 

And I'm not sure, but am I preventing the installation of the other candidates if a single fail occur? 

If that's true: I'm doing the same for the next case. So if it's wrong, it should be changed there as well.
   ```
		2)
			echo ""				echo ""
			__sdkman_echo_red "Stop! Could not get remote version of ${candidate}"				__sdkman_echo_stop "Could not get remote version of ${candidate}"
			return 1				return 1
			;;				;;
   ```

If that's not true: I believe we should add a new test case to ensure the installation of one candidate succeeds despite of the fail of another candidate.

Reply:
I agree.